lesson17/index.js

```diff
@@ -0,0 +1,179 @@
+const url = "https://myjson.dit.upm.es/api/bins/8b9d";
+const main = document.querySelector("main");
+const sliderDirections = ["prev", "next"];
+
+const controlSlide = (btnType) => {
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+
+  if(btnType === "prev") {
+    // 今のcurrentのスライドの一つ前の要素をcurrentにする
+    const prevSlide = currentSlide.previousElementSibling;
+    currentSlide.setAttribute("data-hidden","true");
+    prevSlide.setAttribute("data-hidden","false");
+  } else {
+    // 今のcurrentのスライドの一つ次の要素をcurrentにする
+    const nextSlide = currentSlide.nextElementSibling;
+    currentSlide.setAttribute("data-hidden","true");
+    nextSlide.setAttribute("data-hidden","false");
+  }
+  controlBtnBehavior();
+  updateFractionNum();
+}
```

Comment:
IMO
```suggestion
const controlSlide = (direction) => {
  const currentSlide = document.querySelector('[data-hidden="false"]');
  const showSlide = currentSlide[direction];
  currentSlide.setAttribute("data-hidden", "true");
  showSlide.setAttribute("data-hidden", "false");
  controlBtnBehavior();
  updateFractionNum();
};
```

- nextElementSibling
- previousElementSibling

The process was the same for everything except for these words, so I changed this one to this
```js
const sliderDirections = ["previous", "next"]; 
```
Then we change the return value of setBtnEvent to this.
`${eventTargetBtntype}ElementSibling`
```js

const setBtnEvent = () => {
  const buttons = document.querySelectorAll(".sliderBtn");
  buttons.forEach((button) => {
    button.addEventListener("click", (e) => {
      const eventTargetBtntype = e.target.getAttribute("data-btntype");
      controlSlide(`${eventTargetBtntype}ElementSibling`);
    });
  });
};
```
CSS and other functions have been changed accordingly.

[sandbox](https://codesandbox.io/s/lesson17-forked-1h8m57?file=/index.js)

Reply:
[f8eae0d](https://github.com/hippopo-moo/js-handson-work/pull/19/commits/f8eae0d5a081a03ec1056a7d446f2207a000ccd5)

I had no idea about this method!!!
Thank you for the new awareness!
Thanks to Monaka's IMO, I now recognize that I am not good at rewriting functions, and I will be conscious of writing such code from now on!

`const showSlide = currentSlide[direction];`
↑
Especially this.